lib/install/deps.js

```diff
@@ -643,8 +643,21 @@ var earliestInstallable = exports.earliestInstallable = function (requiredBy, tr
   // if this tree location requested the same module then we KNOW it
   // isn't compatible because if it were findRequirement would have
   // found that version.
-  var deps = tree.package.dependencies || {}
-  if (!tree.removed && requiredBy !== tree && deps[pkg.name]) {
+  var deps = Object.keys(tree.package.dependencies || {})
+
+  // if we're at the root of the tree, and are planning to install
+  // devDependencies in a future pass, don't put any conflicting
+  // packages in the root during the first pass.
+  if (!tree.parent && (npm.config.get('dev') || !npm.config.get('production'))) {
+    // if the same package exists in the root as a devDep, reuse it if the
+    // version matches exactly.
+    if (pkg.version && pkg.version === tree.package.devDependencies[pkg.name]) {
```

Comment:
This can't work because `tree.package.devDependencies[pkg.name])` is a specifier, not a version. This needs to be a semver check.